drivers/serial/uart_nrfx_uarte.c

```diff
@@ -1540,6 +1540,7 @@ static void uarte_nrfx_set_power_state(const struct device *dev,
 			nrfx_timer_enable(&get_dev_config(dev)->timer);
 		}
 		if (get_dev_data(dev)->async) {
+			data->pm_state = new_state;
```

Comment:
Why not just move `data->pm_state = new_state;` up to the top of this block, rather than having it here as well asthe bottom?  There's no control path that depends on the original value, nor that should be inspecting `pm_state`.

Reply:
done